lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -0,0 +1,14 @@
+#include "triton/Target/LLVMIR/ModuleTranslation.h"
+
+std::unique_ptr<llvm::Module> mlir::triton::translateModuleToLLVMIR(
+    Operation *module, llvm::LLVMContext &llvmContext, llvm::StringRef name) {
+  auto llvmMod = mlir::translateModuleToLLVMIR(module, llvmContext, name);
+
+  // Workaround for MLIR not exposing a way to add NoReturn to a function
+  // declaration
+  auto assertfail = llvmMod->getFunction("__assertfail");
+  if (assertfail) {
+    assertfail->addFnAttr(llvm::Attribute::NoReturn);
```

Comment:
should we try to upstream support for this attribute? It looks like it would be fairly simple and would save the workaround (at least on the long term)?

Reply:
Absolutely, an upstream fix to MLIR's llvm dialect would be the best long-term solution. In the short-term I think this workaround is quite self-contained and reduced the number of cross-project interactions needed to fix the inductor issue.

Reply:
Fair enough. Any chance you can kick off a PR for MLIR upstream so that we can reference it in the code?

Reply:
Ot if you were not planning to do the upstream MLIR patch, let me know and I can do it